### ISSUE TITLE
New feature request: split tensor into arbitrary contiguous groups along the specified dimension

### ISSUE THREAD

**Issue by becauseofAI** [apache/incubator-mxnet#15750]
If we want to split a tensor into arbitrary contiguous groups along the specified dimension, such as spliting (N, 80, H, W) to (N, 32, H, W) (N, 32, H, W) (N, 16, H, W), how can we easily achieve this goal with MXNet?

Although the MXNet has provided the API of ```mxnet.ndarray.split(data=None, num_outputs=_Null, axis=_Null, squeeze_axis=_Null, out=None, name=None, **kwargs)```, the ```mxnet.ndarray.split()``` can only split the specified dimension equally and also is limited by the fact that the specified dimension of the ```data``` needs to be divisible by ```num_output```.

And the MXNet also has provided the API of ```mxnet.ndarray.slice_axis(data=None, axis=_Null, begin=_Null, end=_Null, out=None, name=None, **kwargs)```, but the ```mxnet.ndarray.slice_axis()``` can only split one group at a time from the ```begin``` to the ```end```. Although we can implement a function through wraping ```mxnet.ndarray.slice_axis()```  to achieve the goal of arbitrary contiguous groups split, it is not very efficient and friendly.

In fact, the **split** API of Pytorch, TensorFlow, and Numpy can directly implement the above feature. For example, Pytorch's python API can easily achieve the above feature by ```torch.split(x, [32, 16, 16], dim=1)```.

So I think that the MXNet's **split** API of  ```mxnet.ndarray.split()``` requires a more general implementation to  easily achieve the requirement of **spliting a tensor into arbitrary contiguous groups along the specified dimension**.

Looking forward to official attention and updates.

**Comment by vrakesh** [comment]
@mxnet-label-bot add [Feature Request]
